packages/manager/modules/pci/src/projects/project/ai-dashboard/components/ai-pipeline/ai-pipeline.controller.js

```diff
@@ -0,0 +1,78 @@
+import { TOOLS_STATUSES } from './ai-pipeline.constants';
+import exploreImage from './assets/explore.png';
+import serveImage from './assets/serve.png';
+import stockImage from './assets/stock.png';
+import trainImage from './assets/train.png';
+
+export default class AIPipelineCtrl {
+  /* @ngInject */
+  constructor($scope) {
+    this.$scope = $scope;
+  }
+
+  $onInit() {
+    // Compute statuses when items change
+    this.$scope.$watch(
+      '$ctrl.aiItems',
+      () => {
+        this.updateStatus();
+      },
+      true,
+    );
+    // check if user has already ordered ai product.
+    // does not mean that there will be running or stopped status.
+    this.hasData =
+      this.aiItems &&
+      Object.values(this.aiItems).reduce(
+        (acc, itemArray) => acc + itemArray.length,
+        0,
+      ) > 0;
```

Comment:
**nitpick:** This logic is duplicated in more than 1 place. This can be made as a function and reuse it.

Reply:
Totally, I moved it in a function that is used each time it is needed: https://github.com/ovh/manager/pull/8146/commits/b0c18abdaf05ff7256519b7a13d1fd425a784bb4